qiskit/circuit/quantumcircuit.py

```diff
@@ -961,7 +986,7 @@ def compose(
 
         if front:
             # adjust new instrs before original ones and update all parameters
-            mapped_instrs += dest.data
+            mapped_instrs.extend(dest._data)
```

Comment:
tbf, if we implemented `__concat__` and `__inplace_concat__` for `CircuitData`, we could avoid these changes.  No real need to worry about it though, I think.

Reply:
Fair. We may need to do that if we eventually get rid of the `QuantumCircuitData` wrapper. I didn't bother for now since public access of `QuantumCircuit.data` happens through that wrapper, which supports this.